cpp/generate.rb

```diff
@@ -734,12 +792,42 @@ def blockFinderFor( name:"", tag_as:"", start_pattern:nil, needs_semicolon: true
                 /\[/.maybe(@spaces).then(/\]/).maybe(@spaces),
             ).lookAheadFor(/,|\)|\n/)
         )
-    cpp_grammar[:function_definition] = PatternRange.new(
-        tag_as: "meta.function.definition.parameters",
-        start_pattern: avoid_invalid_function_names.then(look_ahead_for_function_name),
-        end_pattern: lookBehindFor(/\)/),
-        includes: [ :parameter_struct, :function_context_c ]
-        )
+        
+    cpp_grammar[:function_definition] = blockFinderFor(
+        name:"function.definition",
+        tag_as:"meta.function.definition",
+        start_pattern: newPattern(
+            qualified_type.then(@spaces.or(ref_deref_definition_pattern)).then(
+                cpp_grammar[:scope_resolution_function_definition]
+            ).then(
+                avoid_invalid_function_names
+            ).then(
+                match: variable_name_without_bounds,
+                tag_as: "entity.name.function.definition"
+            ).maybe(@spaces).lookAheadFor(/\(/)
+        ),
+        needs_semicolon: false,
+        primary_includes: [],
```

Comment:
repeating `primary_includes: []` is just noise, same for `secondary_includes`

Reply:
Ah thanks, I pasted all the inputs to fill them out and I forgot to clean it up afterwords.